common/remote/hooks/azure/azurefile.cpp

```diff
@@ -0,0 +1,827 @@
+
+    return new AzureFileReadIO(this, readStats);
+}
+
+IFileIO * AzureFile::createFileWriteIO()
+{
+    return new AzureFileAppendBlobWriteIO(this);
+}
+
+void AzureFile::ensureMetaData()
+{
+    CriticalBlock block(cs);
+    if (haveMeta)
+        return;
+
+    gatherMetaData();
+    haveMeta = true;
+}
+
+void AzureFile::gatherMetaData()
+{
+    CCycleTimer timer;
+
+    auto client = getClient();
+    azure::storage_lite::blob_client_wrapper wrapper(client);
+
+    auto properties = wrapper.get_blob_property(containerName, blobName);
+    if (errno == 0)
+    {
+        fileExists = true;
+        fileSize = properties.size;
+        blobModifiedTime = properties.last_modified;    // Could be more accurate
+        //MORE: extract information from properties.metadata
+    }
+    else
+    {
+        fileExists = false;
+    }
+
+}
+
+
+bool AzureFile::remove()
+{
+    auto client = getClient();
+    azure::storage_lite::blob_client_wrapper wrapper(client);
+
+    wrapper.delete_blob(containerName, blobName);
+    return (errno == 0);
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+static IFile *createAzureFile(const char *azureFileName)
+{
+    return new AzureFile(azureFileName);
+}
+
+
+//---------------------------------------------------------------------------------------------------------------------
+
+class AzureFileHook : public CInterfaceOf<IContainedFileHook>
+{
+public:
+    virtual IFile * createIFile(const char *fileName)
+    {
+        if (isAzureFileName(fileName))
+            return createAzureFile(fileName);
+        else
+            return NULL;
+    }
+
+protected:
+    static bool isAzureFileName(const char *fileName)
+    {
+        if (!startsWith(fileName, azureFilePrefix))
+            return false;
+        const char * filename = fileName + strlen(azureFilePrefix);
+        const char * slash = strchr(filename, '/');
+        if (!slash)
+            return false;
+        return true;
+    }
+} *azureFileHook;
+
+static CriticalSection *cs;
+
+extern AZURE_FILE_API void installFileHook()
+{
+    CriticalBlock b(*cs); // Probably overkill!
+    if (!azureFileHook)
+    {
+        azureFileHook = new AzureFileHook;
+        addContainedFileHook(azureFileHook);
+    }
+}
+
+extern AZURE_FILE_API void removeFileHook()
+{
+    if (cs)
+    {
+        CriticalBlock b(*cs); // Probably overkill!
+        if (azureFileHook)
+        {
+            removeContainedFileHook(azureFileHook);
+            delete azureFileHook;
+            azureFileHook = NULL;
+        }
+    }
+}
+
+MODULE_INIT(INIT_PRIORITY_STANDARD)
+{
+    cs = new CriticalSection;
+    azureFileHook = NULL;  // Not really needed, but you have to have a modinit to match a modexit
+    return true;
+}
+
+MODULE_EXIT()
+{
+    if (azureFileHook)
+    {
+        removeContainedFileHook(azureFileHook);
+        azureFileHook = NULL;
+    }
+    ::Release(azureFileHook);
```

Comment:
trivial: move into condition above.

Reply:
Where it currently is, azureFileHook will always be null when you call release.

Reply:
copied from archive code.  I'll fix.